plugins/ckan_operators/resource_operator.py

```diff
@@ -9,53 +9,186 @@
         self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
 
     def _resource_exists(self):
-        package = self.ckan.action.package_show(id=self.package_id)
+        # get the package id from a task if its been provided from a task
+        if self.package_name_or_id_task_id and self.package_name_or_id_task_key:
+            self.package_name_or_id = ti.xcom_pull(task_ids=self.package_name_or_id_task_id)[self.package_name_or_id_task_key]
+
+        package = self.ckan.action.package_show(id=self.package_name_or_id)
 
         for r in package["resources"]:
-            if r["name"] == self.resource_name:
+            # get resource name if provided from another task
+            if self.resource_name_task_id and self.resource_name_task_key:
+                self.resource_name = ti.xcom_pull(task_ids=self.resource_name_task_id)[self.resource_name_task_key]
+
+            # find which resource in the package matches the input name
+            # this search ignores suffixes to the resource name, like date ranges
+            if r["name"] == self.resource_name or r["name"][:len(self.resource_name)] == self.resource_name:
                 self.resource = r
                 return True
+             
 
         return False
 
     def execute(self, context):
+        # get resource id from a task if its provided by another task
+        if self.resource_id_task_id and self.resource_id_task_key:
+            self.resource_id = ti.xcom_pull(task_ids=self.resource_id_task_id, key=self.resource_id_task_key)
+        
         if self.resource_id is not None:
+            logging.info("Resource ID used to get resource")
             resource = self.ckan.action.resource_show(id=self.resource_id)
         elif self._resource_exists():
+            logging.info("Resource found using the package id {} and resource name {}". format(self.package_name_or_id, self.resource_name) )
             resource = self.resource
-        else:
+        else: 
+            logging.info("Resource not found - creating a resource called {} in package {}".format(self.resource_name, self.package_name_or_id))
             resource = self.ckan.action.resource_create(
-                package_id=self.package_id,
+                package_name_or_id=self.package_name_or_id,
                 name=self.resource_name,
                 **self.resource_attributes,
             )
 
+        logging.info(resource)
+
+        self.resource_id = resource["id"]
+        logging.info("Resource id: " + self.resource_id)
+
+        # # write the resource id to an input filepath, if the filepath is given
+        # if self.resource_id_filepath and self.resource_id:
+        #     logging.info("Writing resource id to " + str(self.resource_id_filepath))
+        #     f = open( self.resource_id_filepath, "w")
+        #     f.write( self.resource_id )
+
+
         logging.info(f"Returning: {resource}")
         return resource
 
+class EditResourceMetadataOperator(BaseOperator):
```

Comment:
Could you help me understand the use case of this operator, especially when compared to the `ResourceAndFileOperator`? 

I understand one of the limits of the latter is that it only updates the `last_modified` timestamp of the resource to match a file, which I believe is the main scenario for editing resource metadata via a scheduled DAG.

Having trouble picturing a scenario where we'd want to update other attributes in the resource metadata on an ongoing basis.

Reply:
youre right - we dont need to update the other attributes on an ongoing basis eh

I made this because it allowed me to update resource names, which the police data use case demanded.

IYO should I delete this operator and use the `ResourceAndFileOperator`?

Reply:
Let's explore consolidation in the future, if possible, to avoid confusion. Nothing major, I can see the potential use case for this one as well.